### PR TITLE
lambda.part1.exercise1 (Usov)

### PR DESCRIPTION
**Pull request by Sadredneck**

### CODE REVIEW

**Review by Andrewisnew — APPROVED** [review]

File: src/test/java/lambda/part1/exercise/Exercise1.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Person person = FluentIterable
+            .from(persons)
+            .firstMatch(new Predicate<Person>() {
```
Comment (shaart) [inline src/test/java/lambda/part1/exercise/Exercise1.java]: As I understand exercise, here should not be anonymous class, you must declare class which implements Guava ```Predicate``` and use it's instance.

File: src/test/java/lambda/part1/exercise/Exercise1.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Exercise1 {
+
+    private Comparator<Person> personComparator = new Comparator<Person>() {
```
Comment (shaart) [inline src/test/java/lambda/part1/exercise/Exercise1.java]: I think, it's better to put this to the method, where it's used. Same thing to another ```private Comparator<Person>```.

File: src/test/java/lambda/part1/exercise/Exercise1.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            @Override
+            public boolean apply(Person person) {
+                return person.getAge() == 30;
+            }
+        };
+
+        Person person = FluentIterable
+            .from(persons)
+            .firstMatch(personPredicate)
+            .get();
```
Comment (shaart) [inline src/test/java/lambda/part1/exercise/Exercise1.java:114]: It's better to use ```orNull()``` or ```or(defaultValue)``` if you don't like to catch exceptions like ```NoSuchElementException```.

File: src/test/java/lambda/part1/exercise/Exercise1.java (hunk 1 above)
Comment (Sadredneck) [reply src/test/java/lambda/part1/exercise/Exercise1.java]: Fixed

File: src/test/java/lambda/part1/exercise/Exercise1.java (hunk 2 above)
Comment (Sadredneck) [reply src/test/java/lambda/part1/exercise/Exercise1.java]: Ok, fixed

File: src/test/java/lambda/part1/exercise/Exercise1.java (hunk 3 above)
Comment (Sadredneck) [reply src/test/java/lambda/part1/exercise/Exercise1.java:114]: Yep, fixed

**Review by shaart — APPROVED** [review]
Good job!

**Pull request merged**